src/tiledbsoma_ml/dataset.py

```diff
@@ -5,218 +5,619 @@
 
 from __future__ import annotations
 
-from typing import Iterator, Sequence, Tuple
-
-from somacore import ExperimentAxisQuery
+import gc
+import logging
+import os
+import time
+from contextlib import nullcontext
+from math import ceil
+from typing import ContextManager, Iterator, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import torch
+from scipy import sparse
+from tiledbsoma import (
+    DataFrame,
+    Experiment,
+    ExperimentAxisQuery,
+    IntIndexer,
+    SparseNDArray,
+)
 from torch.utils.data import IterableDataset
 
-from tiledbsoma_ml.pytorch import Batch, ExperimentAxisQueryIterable
+from tiledbsoma_ml._csr import CSR_IO_Buffer
+from tiledbsoma_ml._distributed import get_distributed_world_rank, get_worker_world_rank
+from tiledbsoma_ml._experiment_locator import ExperimentLocator
+from tiledbsoma_ml._utils import EagerIterator, batched, splits
+from tiledbsoma_ml.common import Batch, NDArrayJoinId, NDArrayNumber
+
+logger = logging.getLogger("tiledbsoma_ml.dataset")
+
 
+class ExperimentDataset(IterableDataset[Batch]):  # type: ignore[misc]
```

Comment:
Suggest confirming with @aaronwolen that is is good with the name.  Previously we made a conscious decision to be explicit about the `Iterable` nature of the dataset e.g., `ExperimentIterableDataset`. 

I'm agnostic - so whatever you guys decide is good by me.

Reply:
I am also agnostic - I am just requesting you confirm with Aaron, as he previously had a preference